File: Python/Collatz Conjecture.py
def steps(number):
    count = 0
    if number < 1:
        raise ValueError("Only positive numbers are allowed")

    while number != 1:
        count += 1
        if number % 2 == 0:
            number = number / 2
        else:
           number = (3 * number) + 1
    return count

File: C/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start)
{
    if (start <= 0) return -1;
    int steps = 0;

    while (start != 1)
    {
        steps += 1;
        if (start % 2 == 0) start /= 2;
        else start = (start * 3) + 1;
    }

    return steps;
}